dataplane/forwarding/fwdbuilder/action.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package fwdbuilder contains builders for varius proto types.
+package fwdbuilder
+
+import (
+	"encoding/binary"
+
+	fwdpb "github.com/openconfig/lemming/proto/forwarding"
+)
+
+type actionDescBuilder interface {
+	set(*fwdpb.ActionDesc)
+	atype() fwdpb.ActionType
```

Comment:
maybe it's better to be verbose here? actionType()?

Reply:
done